src/server/scripts/Northrend/Naxxramas/boss_gluth.cpp

```diff
@@ -105,52 +129,340 @@ class boss_gluth : public CreatureScript
+        }
+
+        bool Load()
+        {
+            return GetCaster() && GetCaster()->GetEntry() == NPC_GLUTH;
+        }
+    };
+
+    SpellScript* GetSpellScript() const
+    {
+        return new spell_gluth_decimate_SpellScript();
+    }
+
+};
+
+// used by both 28239 & 28404 (single target and aoe zombie-kill spell) to heal Gluth on each target hit.
+
+class spell_gluth_zombiechow_search : public SpellScriptLoader
+{
+public:
+    spell_gluth_zombiechow_search() : SpellScriptLoader("spell_gluth_zombiechow_search") { }
+
+    class spell_gluth_zombiechow_search_SpellScript : public SpellScript
+    {
+        PrepareSpellScript(spell_gluth_zombiechow_search_SpellScript);
+
+        void HealForEachTargetHit()
+        {
+            GetCaster()->ModifyHealth(int32(GetCaster()->CountPctFromMaxHealth(5)));
+        }
+
+        void Register() override
+        {
+            AfterHit += SpellHitFn(spell_gluth_zombiechow_search_SpellScript::HealForEachTargetHit);
+        }
+
+        bool Load()
+        {
+            return GetCaster() && GetCaster()->GetEntry() == NPC_GLUTH;
+        }
+    };
+
+    SpellScript* GetSpellScript() const
+    {
+        return new spell_gluth_zombiechow_search_SpellScript();
+    }
+
+};
+
+// creature 16360 (10man) / 30303 (25man)
+class npc_zombie_chow : public CreatureScript
+{
+public:
+
+    npc_zombie_chow() : CreatureScript("npc_zombie_chow") { }
+
+    struct npc_zombie_chowAI : public ScriptedAI
+    {
+        npc_zombie_chowAI(Creature* creature) : ScriptedAI(creature)
+        {
+            GluthGUID = creature->GetInstanceScript()->GetGuidData(DATA_GLUTH);
+
+            DoCast(me, SPELL_INFECTED_WOUND);
+            state = STATE_ZOMBIE_NORMAL;
+        }
+
+        void UpdateAI(uint32 diff) override
+        {
+            if (!UpdateVictim())
+                return;
+
+            if (state == STATE_ZOMBIE_DECIMATED)
+            {
+                timer += diff;
+                Creature* gluth = ObjectAccessor::GetCreature(*me, GluthGUID);
+                // Putting this in the UpdateAI loop fixes an issue where death gripping a decimated zombie would make the zombie stand still until the rest of the fight.
+                // Also fix the issue where if one or more zombie is rooted when decimates hits (and MovePoint() is called), the zombie teleport to the boss. pretty weird behavior.
+                if (timer>1600 && me->GetExactDist2d(gluth) > 10.0 && me->CanFreeMove()) // it takes about 1600 ms for the animation to cycle. This way, the animation looks relatively smooth.
                 {
-                    me->Kill(me->GetVictim());
-                    me->ModifyHealth(int32(me->CountPctFromMaxHealth(5)));
+                    me->GetMotionMaster()->MovePoint(0, gluth->GetPosition()); // isn't dynamic. So, to take into account Gluth's movement, it must be called periodicly.
```

Comment:
It's not working. I'm not sure why. Possibly because Gluth and the zombies are seen as "friendly" to each other faction wise.